### PR TITLE
Refactor evaluation of Visibility questions and add getting of Eligibility questions

### PR DESCRIPTION
**Pull request by shanemc-goog**
### Description

Adds lookup for which questions are available for eligibility conditions for a given block.

refactors the logic used to make the same determination for visibility so that the main method effectively finds all visible questions up to and including a given block.  Visibility then looks for the questions available through the block before it, and eligibility through its block.

## Release notes

No-op Refactor of logic to determine which questions are available to Visibility conditions to support the in-development eligibility feature.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/application.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://docs.civiform.us/contributor-guide/developer-guide/feature-flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #<issue_number>; Fixes #<issue_number>...

### CODE REVIEW

File: server/app/services/program/ProgramDefinition.java [hunk 1]
```diff
@@ -487,9 +487,42 @@ public ImmutableList<BlockDefinition> getNonRepeatedBlockDefinitions() {
...
+   * Returns a list of the question definitions that may be used to define visibility predicates on
+   * the block definition with the id {@code blockId}.
+   *
+   * <p>The available question definitions for predicates satisfy ALL of the following:
+   *
+   * <ul>
+   *   <li>In a block definition that comes sequentially before the given block definition.
+   *   <li>In a block definition that either has the same enumerator ID as the given block
+   *       definition, or has the same enumerator ID as some "parent" of the given block definition.
+   *   <li>Is not an enumerator.
```
Comment (bion) [inline server/app/services/program/ProgramDefinition.java:500]: suggestion (non-blocking): if it's not too much trouble, could you add and implement "Is not a static text question"?

File: server/app/services/program/ProgramDefinition.java [hunk 2]
```diff
@@ -500,10 +533,13 @@ public ImmutableList<BlockDefinition> getNonRepeatedBlockDefinitions() {
    *   <li>Is not an enumerator.
    * </ul>
    */
-  public ImmutableList<QuestionDefinition> getAvailablePredicateQuestionDefinitions(long blockId)
+  private ImmutableList<QuestionDefinition> getAvailablePredicateQuestionDefinitions(long blockId)
```
Comment (bion) [inline server/app/services/program/ProgramDefinition.java:536]: suggestion (blocking): looks like the javadoc for this isn't quite accurate after this change --> "In a block definition that comes sequentially before the given block definition."

**Review by bion — CHANGES_REQUESTED** [review]

File: server/app/services/program/ProgramDefinition.java (hunk 1 above)
Comment (shanemc-goog) [reply server/app/services/program/ProgramDefinition.java:500]: That should be separate from this refactor as it's unrelated to it.

File: server/app/services/program/ProgramDefinition.java (hunk 2 above)
Comment (shanemc-goog) [reply server/app/services/program/ProgramDefinition.java:536]: thanks, c&p error.

I thought it better to make this private which made the doc not technically reference-able in the other methods docs

**Review by bion — APPROVED** [review]

**Pull request merged**